sake/redis.py

```diff
@@ -691,6 +693,55 @@ async def set_user(self, user: users.User, /, *, expire_time: typing.Optional[Ex
         await client.set(int(user.id), self.marshaller.serialize_user(user), pexpire=expire_time)
 
 
+class PrefixCache(ResourceClient, traits.PrefixCache):
+    __slots__: typing.Sequence[str] = ()
+
+    @classmethod
+    def index(cls) -> typing.Sequence[ResourceIndex]:
+        # <<Inherited docstring from ResourceClient>>
+        return (ResourceIndex.PREFIX,)
+
+    def subscribe_listeners(self) -> None:
+        # <<Inherited docstring from sake.traits.Resource>>
+        super().subscribe_listeners()
+
+    def unsubscribe_listeners(self) -> None:
+        # <<Inherited docstring from sake.traits.Resource>>
+        super().unsubscribe_listeners()
+
+    async def clear_prefixes(self) -> None:
+        # <<Inherited docstring from sake.traits.PrefixCache>>
+        client = await self.get_connection(ResourceIndex.PREFIX)
+        await client.flushdb()
+
+    async def delete_prefixes(self, guild_id: snowflakes.Snowflakeish, /) -> None:
+        # <<Inherited docstring from sake.traits.PrefixCache>>
+        client = await self.get_connection(ResourceIndex.PREFIX)
+        await client.delete(int(guild_id))
+
+    async def get_prefixes(self, guild_id: snowflakes.Snowflakeish, /) -> typing.List[str]:
+        # <<Inherited docstring from sake.traits.PrefixCache>>
+        guild_id = int(guild_id)
+        client = await self.get_connection(ResourceIndex.PREFIX)
+        data = await client.get(guild_id)
+        if not data:
+            raise errors.EntryNotFound(f"Prefix entry `{guild_id}` not found")
+
+        return self.marshaller.deserialize_prefixes(data)
+
+    def iter_prefixes(self, *, window_size: int = WINDOW_SIZE) -> traits.CacheIterator[typing.List[str]]:
+        # <<Inherited docstring from sake.traits.PrefixCache>>
+        return redis_iterators.Iterator(
+            self, ResourceIndex.PREFIX, self.marshaller.deserialize_prefixes, window_size=window_size
+        )
```

Comment:
When sset is used then we'll want to have an accompanying `add_prefix` or `add_prefixes` my method